index.ts

```diff
@@ -1,31 +1,51 @@
 import { multiUrlGenerator } from './lib/multiUrlGenerator';
-import fetch from 'node-fetch';
+import fetch, { Response } from 'node-fetch';
+import console from 'console';
+import readline from 'readline';
+const URL_CONFIG_PATH = './config/url_components';
+const PARSER_PATH = './config/parsers/';
+type ScraperMode = 'wikipedia' | 'mojang';
+const MODES: ReadonlyArray<ScraperMode> = ['wikipedia', 'mojang'];
 
-// Test fetching Wiki articles
-const requestUrls = multiUrlGenerator(
-  'https://$lang.wikipedia.org/wiki/$article',
-  [
-    { $lang: 'en', $article: 'Plato' },
-    { $lang: 'es', $article: 'Aristóteles' },
-  ]
-);
+const client = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
 
-async function main() {
-  for (const url of requestUrls) {
+client.question('Choose the mode you want to run the program ', (MODE) => {
+  //TODO: change the question method to setPrompt()/popPromt() or reformulate the question() string to ask an actual question.
+  if (!MODES.includes(MODE as ScraperMode)) {
+    console.error(
+      `Specified scraper mode "${MODE}" not found in config files.`
+    );
+    process.exit(0);
+  }
+  main(MODE);
+  client.close();
+});
+
+async function getUrls(configFileName: string): Promise<string[]> {
+  const urlConfig = await import(`${URL_CONFIG_PATH}/${configFileName}`);
```

Comment:
While you're at it, let's add some basic validation here:

```typescript
async function getUrls(configFileName: string): Promise<string[]> {
  const urlConfig = await import(`${URL_CONFIG_PATH}/${configFileName}`);
  if (urlConfig.baseTemplate && urlConfig.components) {
    return multiUrlGenerator(urlConfig.baseTemplate, urlConfig.components);
  }
  console.error('Wrong url config format');
  process.exit(0);
}
```

Reply:
👍